Android/app/src/main/res/layout/fragment_funfact.xml

```diff
@@ -51,4 +51,31 @@
         android:textColor="#000000"
         android:textSize="25sp" />
 
+    <LinearLayout
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        android:padding="10dp">
+
+        <TextView
+            android:id="@+id/text_source"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/text_source"
```

Comment:
When there is no text source (it is null), the text view will show
**Source : **

This won't look good.

Reply:
It's default value is gone. My bad. Ignore this comment.